src/sdk/namespace_fs.js

```diff
@@ -1419,23 +1419,22 @@ class NamespaceFS {
                 const versioned_path = latest_ver_info && this._get_version_path(key, latest_ver_info.version_id_str);
                 const versioned_info = latest_ver_info && await this._get_version_info(fs_context, versioned_path);
 
-                gpfs_options = is_gpfs ?
-                    await this._open_files_gpfs(fs_context, new_ver_tmp_path, latest_ver_path, upload_file,
-                        latest_ver_info, open_mode, undefined, versioned_info) :
-                    undefined;
                 const bucket_tmp_dir_path = this.get_bucket_tmpdir_full_path();
-                dbg.log1('Namespace_fs._move_to_dest_version:', latest_ver_info, new_ver_info, gpfs_options);
-
-                if (this._is_versioning_suspended()) {
+                if (!is_gpfs && this._is_versioning_suspended()) {
                     if (latest_ver_info?.version_id_str === NULL_VERSION_ID) {
                         dbg.log1('NamespaceFS._move_to_dest_version suspended: version ID of the latest version is null - the file will be unlinked');
-                        await native_fs_utils.safe_unlink(fs_context, latest_ver_path, latest_ver_info,
-                            gpfs_options?.delete_version, bucket_tmp_dir_path);
+                        await native_fs_utils.safe_unlink(fs_context, latest_ver_path, latest_ver_info, undefined, bucket_tmp_dir_path);
                     } else {
                         // remove a version (or delete marker) with null version ID from .versions/ (if exists)
                         await this._delete_null_version_from_versions_directory(key, fs_context);
                     }
                 }
+                gpfs_options = is_gpfs ?
+                await this._open_files_gpfs(fs_context, new_ver_tmp_path, latest_ver_path, upload_file,
+                    latest_ver_info, open_mode, undefined, versioned_info) :
+                undefined;
+                dbg.log1('Namespace_fs._move_to_dest_version:', latest_ver_info, new_ver_info, gpfs_options);
+
                 if (latest_ver_info &&
```

Comment:
Not related to this line, on the github issue there is a test called test_s3cever102, is it already running in our ceph tests run?

Reply:
no this test was made by herald. I think there is already test for that on our test. I'll check and add one if its missing

Reply:
there is already a test in test_bucketspace_versioning.js:
https://github.com/noobaa/noobaa-core/blob/3c8a29a2153fe51e46b859617c8a193758e8abe7/src/test/unit_tests/test_bucketspace_versioning.js#L453-L478

Reply:
there is already a test for this in test_bucketspace_versioning.js:
https://github.com/noobaa/noobaa-core/blob/3c8a29a2153fe51e46b859617c8a193758e8abe7/src/test/unit_tests/test_bucketspace_versioning.js#L366-L372